torch/csrc/jit/codegen/cuda/test/test_gpu_tensorcore_perf.cpp

```diff
@@ -0,0 +1,719 @@
+    FusionGuard fg(&fusion);
+    auto tv0 = makeContigTensor(2, DataType::Half);
+    auto tv1 = makeContigTensor(2, DataType::Half);
+
+    fusion.addInput(tv0);
+    fusion.addInput(tv1);
+
+    auto tv2 = matmul(tv0, tv1, layout);
+
+    fusion.addOutput(tv2);
+
+    MatMulTileOptions gemm_tile;
+    gemm_tile.cta_tile = test_desc.getCtaTileDesc();
+    gemm_tile.warp_tile = test_desc.getWarpTileDesc();
+    gemm_tile.instruction_tile = test_desc.getMmaDesc();
+
+    auto mma_builder =
+        MmaBuilder(test_desc.getMmaOp(), gemm_tile).layout(layout);
+
+    MatmulParam params(mma_builder);
+    params.tile_sizes = gemm_tile;
+    params.async_gmem_load_operands = true;
+
+    const int stages = test_desc.getStages();
+    const bool smem_write = (0 != stages) ? true : false;
+    params.double_buffer_options.double_buffer_smem_write = smem_write;
+    params.double_buffer_options.smem_double_buffer_stage = stages;
+
+    scheduleMatmul(tv2, tv0, tv1, params);
+    FusionExecutor fe;
+    auto needsCompilation = true;
+
+    for (const auto& problem_desc : test_desc.getProblemsDesc()) {
+      printTestCaseDesc(test_desc, problem_desc);
+      try {
+        at::manual_seed(0);
+        auto inputs = fp16MatmulAtInput(
+            problem_desc.m, problem_desc.n, problem_desc.k, layout);
+
+        if (needsCompilation) {
+          NVFUSER_TEST_CUDA_ARCH_COMPILE_CHECK(
+              test_desc.getRawArchMajor(),
+              test_desc.getRawArchMinor(),
+              fe.compileFusion(&fusion, {inputs.first, inputs.second}));
+          needsCompilation = false;
+        }
+
+        for (int32_t i = 0; i < test_desc.getIters(); ++i) {
+          fe.runFusion({inputs.first, inputs.second});
+        }
+
+        auto cg_outputs = fe.runFusion({inputs.first, inputs.second});
+        if (!test_desc.isBenchmark()) {
+          auto tref = atMatmul(
+              inputs.first.to(at::kFloat),
+              inputs.second.to(at::kFloat),
+              layout);
+          TORCH_CHECK(cg_outputs[0].allclose(
+              tref,
+              test_desc.getAbsoluteTolerance(problem_desc),
+              test_desc.getRelativeTolerance(problem_desc)));
+        }
+      } catch (const std::exception& e) {
+        if (test_desc.isBenchmark()) {
+          // NOTE: ignore issues to avoid breaking execution of multiple problem
+          // sizes - exception will kill the run
+        } else {
+          throw;
+        }
+      }
+    }
+  }
+}
+
+// NOTE: simple gemm that uses matmulSchedule, based on:
+//    FusionAmpereMatmul_CUDA
+//    FusionAmpereMatmulPipelineGmem_CUDA
+//    FusionAmpereMatmulLargeLoad_CUDA
+TEST_F(NVFuserTest, perf__FusionAmpereMatmulPipelineGmem_CUDA__regenKernels) {
+  // NOTE:
+  // - set 'PYTORCH_NVFUSER_DUMP=dump_eff_bandwidth' to gather execution time
+  // - the following env variables, with example values, need to be set
+  //   to perform a perf check of generated kernel
+  //   - NVFUSER_CTA=128x128x32
```

Comment:
Probably a good idea to use more specific names to avoid name conflicts. Maybe `NVFUSER_MATMUL_CTA`, `NVFUSER_MATMUL_WARP`, etc?

Reply:
Sure, this makes sense - I wasn't sure what name to use as I don't know for how long this piece of code will be needed.
I will update env vars definitions and tools that use these tests before I move them to benchmark framework.
Thanks!